addons/mrp/models/mrp_production.py

```diff
@@ -1279,7 +1283,7 @@ def _set_qty_producing(self, pick_manual_consumption_moves=True):
 
             new_qty = float_round((self.qty_producing - self.qty_produced) * move.unit_factor, precision_rounding=move.product_uom.rounding)
             move._set_quantity_done(new_qty)
-            if (not move.manual_consumption or pick_manual_consumption_moves) and move.quantity:
+            if (not move.manual_consumption or pick_manual_consumption_moves) and move.quantity and (move.product_id != self.product_id or move.product_id.tracking != 'serial'):
```

Comment:
I know it's an edge case, but wouldn't work on MOs in which the end product (which happens to be a serial) is also a component.